analysis/parser/RawTextParser/src/main/kotlin/de/maibornwolff/codecharta/parser/rawtextparser/MetricCollector.kt

```diff
@@ -43,7 +43,11 @@ class MetricCollector(
 
     private fun parseFile(file: File): FileMetrics {
         val metrics = MetricsFactory.create(metrics, parameters)
-        file.readLines().stream().parallel().forEach { line -> metrics.forEach { it.parseLine(line) } }
+
+        file
+            .useLines { it.toList() }
+            .forEach { line -> metrics.forEach { it.parseLine(line) } }
```

Comment:
I suggest to read the file line by line and to parse those lines immediately instead of collecting all of them first and to start parsing afterwards.

A possible implementation is described here: https://bezkoder.com/kotlin-read-file/

Note: https://github.com/MaibornWolff/codecharta/pull/1286/files#diff-310a91aae2337cc854312494bf4e1e09R34 has an obsolete `it.isFile` check (the sequence is already filtered in Line 28).